### PR TITLE
layernom inplace

### PR DESCRIPTION
**Pull request by Krovatkin**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63437
* #63436

Differential Revision: [D30388824](https://our.internmc.facebook.com/intern/diff/D30388824)

### CODE REVIEW

File: test/jit/test_freezing.py [hunk 1]
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
...
+                        sub_model.eval()
+                        mod = torch.jit.freeze(torch.jit.script(sub_model))
+                        self.run_pass("convert_frozen_ops_to_mkldnn", mod.graph)
+                        # if weight and bias are present and shape is the last dimension
+                        # we should convert `aten::layer_norm` to `prim::MKLDNNLayerNorm`
+                        if layernorm.elementwise_affine and len(param[1]) == 1:
+                            inplace_suffix = "_" if inplace else ""
+                            FileCheck().check("prim::MKLDNNLayerNorm" + inplace_suffix).check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
+                        else:
+                            FileCheck().check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
```
Comment (eellison) [inline test/jit/test_freezing.py:1965]: this is not checking anything, we should be checking that aten::layernorm exists or whatever

File: test/jit/test_freezing.py [hunk 2]
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
...
+                        self.assertTrue(torch.allclose(sub_model(param[2]), mod(param[2]), 1e-04, 1e-04))
+
+    @skipIfNoTorchVision
+    def test_layernorm2(self):
+        with set_default_dtype(torch.float):
+            model = torchvision.models.resnet18()
```
Comment (eellison) [inline test/jit/test_freezing.py:1971]: model is unused

File: test/jit/test_freezing.py [hunk 3]
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
...
+
+                def forward(self, x):
+                    y = self.op(x)
+                    return self.layernorm(y) + y
+
+            N, C, H, W, = 10, 3, 224, 224
+            for param in ((torch.nn.Linear(H, W), [W], torch.randn(N, C, W)),):
+
+                for layernorm in (torch.nn.LayerNorm(param[1]),):
+                    sub_model = ResidualLayernorm(param[0], layernorm)
```
Comment (eellison) [inline test/jit/test_freezing.py:1988]: why do we need this test in addition to what exists above ?

File: test/jit/test_freezing.py [hunk 4]
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
...
+                        # we should convert `aten::layer_norm` to `prim::MKLDNNLayerNorm`
+                        if layernorm.elementwise_affine and len(param[1]) == 1:
+                            inplace_suffix = "_" if inplace else ""
+                            FileCheck().check("prim::MKLDNNLayerNorm" + inplace_suffix).check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
+                        else:
+                            FileCheck().check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
+                        self.assertTrue(torch.allclose(sub_model(param[2]), mod(param[2]), 1e-04, 1e-04))
+
+    @skipIfNoTorchVision
+    def test_layernorm2(self):
```
Comment (eellison) [inline test/jit/test_freezing.py:1969]: can you have a more descriptive testname ? what is this testing that the other test is not

**Review by eellison — APPROVED** [review]

**Comment by Krovatkin** [comment]
@krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30388824).

**Comment by Krovatkin** [comment]
@krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30388824).

**Comment by Krovatkin** [comment]
@krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30388824).

**Pull request closed without merging**